backend/pkg/api/connect/service/clusterstatus/util.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package clusterstatus
+
+import consolev1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/console/v1alpha1"
+
+func setStatus(s *consolev1alpha1.ComponentStatus, status consolev1alpha1.StatusType, reason string) {
+	if status > s.Status {
```

Comment:
I am not sure I understand this `>` check?

Reply:
The status is an enum / iota where a greater number means => worse condition.

1 = Healthy
2 = Degraded
3 = Unhealthy.

Because we only want to return one status + error messsage, we use this comparison to see if the status we are setting is worse than the status we have so far.

Reply:
Looking at the code and context how it's used I get it now... A comment might be helpful.